src/engine/AbsoluteValueConstraint.cpp

```diff
@@ -179,24 +191,38 @@ void AbsoluteValueConstraint::notifyUpperBound( unsigned variable, double bound
          !FloatUtils::lt( bound, getUpperBound( variable ) ) )
         return;
 
-     setUpperBound( variable, bound );
+    setUpperBound( variable, bound );
     // Check whether the phase has become fixed
     fixPhaseIfNeeded();
 
     // Update partner's bound
     if ( isActive() && _boundManager )
     {
+
+        bool proofs = _boundManager->shouldProduceProofs();
+
+        if ( proofs )
+        {
+            createPosTighteningRow();
+            createNegTighteningRow();
+        }
+
         if ( variable == _b )
         {
-            if ( bound > 0 )
+            if ( FloatUtils::gt( bound, 0 ) && ( !proofs || !phaseFixed() ) )
             {
-                double fUpperBound = FloatUtils::max( bound, -getLowerBound( _b ) );
-                _boundManager->tightenUpperBound( _f, fUpperBound );
+                double fUpperBound = FloatUtils::max( bound, -getLowerBound( _b ) ) ;
+                if ( proofs )
+                    _boundManager->addLemmaExplanation( _f, fUpperBound, UPPER, { variable, variable }, UPPER, getType() );
```

Comment:
For the first question - addLemmaExplanation includes calling tightenUpperBound or tightenLowerBound (depending on the case).
For the second - this tightening cannot be explained using a linear combination of the variables. I think this is because the tightening uses information from both phases. It is not the case for the other tightenings.